### PR TITLE
Refactor firstBlockAllowedToMonitor

### PR DESCRIPTION
**Pull request by karlb**
This makes more of the logic publicly accessible, which allows us to
remove duplicated logic from the test setup and the from the MS.

Since the function signature changed, this will need a change in the MS.

### CODE REVIEW

File: raiden_contracts/data/source/services/MonitoringService.sol [hunk 1]
```diff
@@ -227,15 +233,15 @@ contract MonitoringService is Utils {
         );
     }
 
-    function isAllowedToMonitor(
+    function firstBlockAllowedToMonitor(
         TokenNetwork token_network,
```
Comment (karlb) [inline raiden_contracts/data/source/services/MonitoringService.sol]: This can be called publicly by passing in the TokenNetwork's address.

**Comment by palango** [comment]
@karlb Do you want to merge this before the next release?

**Comment by karlb** [comment]
> @karlb Do you want to merge this before the next release?
> 
> If so, please rebase.

There's no need to hurry this PR. I still rebased, so that it can be easily merged when convenient.

**Comment by karlb** [comment]
> Maybe `firstBlockAllowedToMonitor` -> `firstBlockAllowedToMonitorChannel`

Then I'll also use `firstBlockAllowedToMonitor` instead of `pureFirstBlockAllowedToMonitor`. That
* is one less rename in the diff
* a bit shorter
* a more usual name, since we usually don't include information like "pure" in the variable name.

**Pull request merged**